### ISSUE TITLE
Base no-lone-blocks rule: emits an error with `using` keyword

### ISSUE THREAD

**Issue by MillerSvt** [typescript-eslint/typescript-eslint#8682]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.tsx&code=FAb2AJwVwZwSwHYHNwENwF5wGMBOBTVAF3wFkB7bAa31wAoByAM3PIYEoBuYAXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaefXRQgIxIGMBrVDSRaafaScMAXxC6A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
{
  using a = createMocker('foo');
}
```

### ESLint Config

```javascript
{
  "rules": {
    "no-lone-blocks": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

No errors, because using keyword equals const keyword

### Actual Result

Error: Nested block is redudant

### Additional Info

Related to: https://github.com/eslint/eslint/issues/18204#issuecomment-1997901666

**Comment by Josh-Cena** [comment]
This is an ESLint core rule. `using` is a JS feature that ESLint should eventually support. This has nothing to do with TypeScript and I'm in total disagreement with responses in that thread. Sorry if it sounds like we are kicking you around, but there's really nowhere we can help you get around it in ts-eslint.

**Comment by MillerSvt** [comment]
I see several solutions to this problem:
1. Temporary parse these expressions as constants.
2. Write an extension for eslint rules as you have done many times before

**Comment by Josh-Cena** [comment]
> Temporary parse these expressions as constants (which they actually are; maybe with flag isUsing for backward compatibility)

We will not be working around how other rules work; `using` is a declaration completely separate from `const` or `let`. Doing so may lead to other buggy logic downstream.

> Write an extension for eslint rules as you have done many times before

Yeah we could do that, definitely. I see they have raised the same suggestion.

**Comment by bradzacher** [comment]
I'm personally okay closing this.
The intersection of `using` + intentional lone block + `no-lone-blocks` rule is a very, very small subset of codebases.
I don't think it's worth us bringing on the maintenance burden of a new lint rule solely so that we can temporarily mitigate this until the feature reaches stage 4 and eslint core adds support.

**Comment by kirkwaiblinger** [comment]
fun fact - I'm >90% sure that the only thing that would have to change in the base rule right now would be 
```diff
             ruleDef.VariableDeclaration = function(node) {
-                if (node.kind === "let" || node.kind === "const") {
+                if (node.kind === "let" || node.kind === "const" || node.kind === "using") {
                     markLoneBlock(node);
                 }
             };
```

Or, if you wanted to be sneaky about it:
```diff
             ruleDef.VariableDeclaration = function(node) {
-                if (node.kind === "let" || node.kind === "const") {
+                if (node.kind !== "var") {
                     markLoneBlock(node);
                 }
             };
```

**Comment by bradzacher** [comment]
Yeah exactly! But we'd also need to bring in a hundred or so lines to declare the boilerplate of the extension rule, plus a few hundred lines of tests, plus docs.

A one line change isn't as simple as you'd like when releasing an extension rule :(

**Comment by Josh-Cena** [comment]
Is is possible to ask ESLint to change it to `node.kind !== "var"`? No future variable declarations will ever be non-block-based if TC39 is in their right minds.

**Comment by kirkwaiblinger** [comment]
> Is is possible to ask ESLint to change it to `node.kind !== "var"`? No future variable declarations will ever be non-block-based if TC39 is in their right minds.

FWIW `await using` is also coming, and is a concrete example of another problem that would be solved before coming into existence by making that change. Just throwing that out there In case that data point is helpful to someone championing the `node.kind !== "var"` cause.

**Comment by bradzacher** [comment]
> Is is possible to ask ESLint to change it to node.kind !== "var"? 

There's only one way to find out! Anyone can ask in the linked issue or file a new issue asking!

**Comment by Josh-Cena** [comment]
ESLint has decided to go with the approach @kirkwaiblinger proposed 🎉
